compute/disable_usage_export.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_usage_report_disable]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+// disableUsageExport disables Compute Engine usage export bucket for the Cloud Project.
+func disableUsageExport(w io.Writer, projectID string) error {
+	// projectID := "your_project_id"
+	ctx := context.Background()
+	projectsClient, err := compute.NewProjectsRESTClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewProjectsRESTClient: %v", err)
+	}
+	defer projectsClient.Close()
+
+	// Providing an empty UsageExportLocationResource value disables the usage report generation.
+	req := &computepb.SetUsageExportBucketProjectRequest{
+		Project:                     projectID,
+		UsageExportLocationResource: &computepb.UsageExportLocation{},
```

Comment:
If nothing is passed will this also disable the report? Or does an empty struct need to be sent?

Reply:
just tested it, it works without empty struct. (in Node.js for example it doesn't work, you need to pass empty object)